eo-maven-plugin/src/test/java/org/eolang/maven/VerifyMojoTest.java

```diff
@@ -100,19 +105,58 @@ void doesNotDetectWarningWithoutCorrespondingFlag(@TempDir final Path temp) {
     }
 
     @Test
-    @Disabled
-    void detectsCriticalError(@TempDir final Path temp) {
+    void failsOptimizationOnError(@TempDir final Path temp) {
         Assertions.assertThrows(
             IllegalStateException.class,
             () -> new FakeMaven(temp)
                 .withProgram(
-                    "+package f\n",
-                    "[] > main",
-                    "  [] > name",
-                    "  [] > name"
+                    "+package f",
+                    "+alias THIS-IS-WRONG org.eolang.io.stdout\n",
+                    "[args] > main",
+                    "  (stdout \"Hello!\").print > @"
                 )
-                .execute(new FakeMaven.Verify()),
-            "Program with duplicate names should have failed on critical error, but it didn't"
+                .execute(new FakeMaven.Verify())
+        );
+    }
+
+    @Test
+    void failsOnWarning(@TempDir final Path temp) throws Exception {
+        final FakeMaven maven = new FakeMaven(temp)
+            .withProgram(
+                "+architect yegor256@gmail.com",
+                "+tests",
+                "+package org.eolang.examples\n",
+                "[] > main",
+                "  [] > @",
+                "    hello > test"
+            );
+        VerifyMojoTest.applyXsl(
+            "org/eolang/maven/set-warning-severity.xsl",
+            maven.execute(ParseMojo.class)
+                .result()
+                .get("target/1-parse/foo/x/main.xmir")
+        );
+        Assertions.assertThrows(
```

Comment:
@levBagryansky Could you add human readable message for this assertion, please?
https://www.yegor256.com/2023/01/19/layout-of-tests.html#assertions